colearn_grpc/grpc_learner_server.py

```diff
@@ -153,11 +145,13 @@ def ProposeWeights(self, request, context):
         self._learner_mutex.acquire()
         try:
             _logger.debug("Start training...")
-            w = ipb2.Weights()
             weights = self.learner.mli_propose_weights()
-            w.weights = encode_weights(weights)
             _logger.debug("Training done!")
-            yield w
+
+            weights_part_iterator = weights_to_iterator(weights)
```

Comment:
do we wan't to lock on the learner during the whole streaming process?

Reply:
I'm not sure how else to do it? Is there some way of not locking?